ably/rest_client_test.go

```diff
@@ -21,38 +21,39 @@ import (
 
 var _ = Describe("RestClient", func() {
 	var (
-		server *httptest.Server
+		srv *httptest.Server
 
-		createMockedRestClient = func(server *httptest.Server) *http.Client {
+		newHTTPClientMock = func(srv *httptest.Server) *http.Client {
 			return &http.Client{
 				Transport: &http.Transport{
-					Proxy: func(req *http.Request) (*url.URL, error) {
-						return url.Parse(server.URL)
-					},
+					Proxy: func(req *http.Request) (*url.URL, error) { return url.Parse(srv.URL) },
 				},
 			}
 		}
 	)
 
-	Context("with a failing request", func() {
+	Context("with a failing req", func() {
 		var (
-			statusCode int
-			request    *http.Request
+			req    *http.Request
+			cli    *ably.RestClient
```

Comment:
I am with @kouno on this, I don't see the benefit of `srv` of `server` or `req` over `request`.  Whilst perhaps this shorter notation is used in other packages, I would prefer to have our variables descriptive.  `status` is fine, however please note that in our errors we have `status` and `code` where code is our internal error code.